common/cache/domainCache.go

```diff
@@ -597,7 +597,7 @@ func (c *DefaultDomainCache) getDomain(
 	if cacheHit {
 		return c.getDomainByID(id, true)
 	}
-	if err := c.refreshDomainsLocked(); err != nil {
+	if err := c.refreshDomainsLocked(context.Background()); err != nil {
```

Comment:
Not as part of this PR but we should update DefaultDomainCache to have a ctx & cancel so that Stop can cancel the context to honor Stop signal